centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -780,13 +780,39 @@ object Operations extends StrictLogging {
       }
 
       def checkArchived(): IO[Boolean] = for {
-        archiveStatus <- CentaurCromwellClient.archiveStatus(workflowId)
+        archiveStatus <- CentaurCromwellClient.archiveStatus(submittedWorkflow.id)
         isArchived <- validateMetadataArchiveStatus(archiveStatus)
       } yield isArchived
 
+      def validateMetadataSourceArchived(): IO[Boolean]  = for {
+        metadataSource <- CentaurCromwellClient.metadataWithId(submittedWorkflow.id).map(_.asFlat.stringifyValues.get("metadataSource"))
+        isMetadataSourceArchived <- {
+          if (metadataSource == Option(JsString("Archived"))) {
```

Comment:
TOL could be a `metadataSource match {...`

Reply:
I think `if (metadataSource.contains(JsString("Archived")))` is the "idiomatic scala" way to express this